compiler/rustc_hir_analysis/src/check/wfcheck.rs

```diff
@@ -431,6 +431,8 @@ fn check_gat_where_clauses(tcx: TyCtxt<'_>, trait_def_id: LocalDefId) {
                         )
                     }
                     ty::AssocKind::Const { .. } => None,
+                    // This does not change judgement on GAT
+                    ty::AssocKind::AutoImpl => continue,
```

Comment:
I don't want the type system to have to care about a new assoc kind. Also please get types approval before having this merged (I've assigned myself). Why is this PR doing more than just parser stuff like the title and PR description indicates?